### PR TITLE
CouchDB 3.2 release notes

### PR DESCRIPTION
**Pull request by wohali**
You can more easily read the RST content at this link:

https://github.com/apache/couchdb-documentation/blob/186c0896222a55817dbd8fdcaabbf30b56935b1d/src/whatsnew/3.2.rst

### CODE REVIEW

File: src/whatsnew/3.2.rst [hunk 1]
```diff
@@ -57,4 +113,140 @@ Features and Enhancements
...
+* :ghissue:`3212`: Base and extra headers are properly combined when making replicator
+  requests that contain duplicate headers.
+* :ghissue:`3201`: When using a POST with request body to pass parameters to a view-like
+  request, the boolean parameters are accepting only JSON strings, but not booleans.
+  Now, CouchDB accepts ``true`` and ``false`` for the ``stable`` parameter, in addition
+  to ``"true"`` and ``"false"``.
+  comment in
+* :ghissue:`1988`: Attachment operations ``PUT /db/doc`` and ``POST /db`` now perform
+  consistent attachment name validation.
+* :ghissue:`3249`: Documents with lots of conflicts no longer blow up couchjs if the
```
Comment (jaydoane) [inline src/whatsnew/3.2.rst:187]: s/couchjs/couch_js/?

File: src/whatsnew/3.2.rst [hunk 2]
```diff
@@ -57,4 +113,140 @@ Features and Enhancements
...
+* :ghissue:`3255`, :ghissue:`2558`: View files that have incorrect ``db_headers``
+  now reset the index forcing a rebuild.
+* :ghissue:`3271`: Attachments that are stored uncompressed but later replicated to
+  nodes that compress the attachment no longer fail an internal md5 check that would
+  break eventual consistency between nodes.
+* :ghissue:`3277`: ``req_body`` requests that have ``req_body`` set already now
+  properly return the field without parsing.
+* :ghissue:`3279`: Some default headers were missing from some responses in replication,
+  including ``X-CouchDB-Body-Time`` and ``X-Couch-Request-ID``.
+* :ghissue:`3329`, :ghissue:`2962`: CouchDB no longer returns broken couchjs processes
```
Comment (jaydoane) [inline src/whatsnew/3.2.rst:202]: couch_js?

Comment (wohali) [reply src/whatsnew/3.2.rst:202]: Is it the Erlang process that's broken, or the external process? If the external, the binary is called `couchjs`. If the former, `couch_js.

File: src/whatsnew/3.2.rst (hunk 1 above)
Comment (wohali) [reply src/whatsnew/3.2.rst:187]: See other comment.

File: src/whatsnew/3.2.rst [hunk 3]
```diff
@@ -57,4 +113,140 @@ Features and Enhancements
...
+* :ghissue:`3506`, :ghissue:`3416`, :ghissue:`3377`: CouchDB now provides a Prometheus
+  compatible endpoint at ``GET /_node/{node-name}/_prometheus``. A configuration option
+  allows for scraping via a different port (17986) that does not require authentication,
+  if desired. More information is available at the :ref:`Prometheus API endpoint
+  <api/server/prometheus>` summary.
+
+* :ghissue:`3697`, :issue:`883` (JIRA): As an opt-in policy, CouchDB can now stop
+  encoding the plus sign ``+`` in non-query parts of URLs, in compliance with the
+  original CouchDB standards. The opt-in is via the ``[chttpd] decode_plus_to_space =
+  true`` setting. *In CouchDB 4.x, this is an opt-out policy.*
```
Comment (janl) [inline src/whatsnew/3.2.rst:125]:
```suggestion
  true`` setting. *In CouchDB 4.x, this is going to be an opt-out policy.*
```

File: src/whatsnew/3.2.rst [hunk 4]
```diff
@@ -57,4 +113,140 @@ Features and Enhancements
...
+* :ghissue:`3700`: The ``maybe_close`` message is now sent to the correct internal
+  process.
+* :ghissue:`3183`: The smoosh operator guide now recommends to use the ``rpc:multicall``
+  function.
+* :ghissue:`3712`: Including a payload within a ``DELETE`` operation no longer hangs
+  the next request made to the same mochiweb acceptor.
+* :ghissue:`3715`: For clusters with databases where ``n > [cluster] n``, attachments
+  chunks are longer dropped on quorum writes.
+* :ghissue:`3507`: If a file is truncated underneath CouchDB, CouchDB will now log
+  if it finds this situation with a ``file_truncate_error``.
```
Comment (janl) [inline src/whatsnew/3.2.rst:245]:
```suggestion
  the filename if it finds this situation with a ``file_truncate_error``.
```

**Review by janl — APPROVED** [review]
amazing as always, also, this is a strong release, congrats everybody :)

left two nitpicks

**Review by nickva — APPROVED** [review]
Well done

File: src/whatsnew/3.2.rst [hunk 5]
```diff
@@ -57,4 +113,140 @@ Features and Enhancements
...
+  encoding the plus sign ``+`` in non-query parts of URLs, in compliance with the
+  original CouchDB standards. The opt-in is via the ``[chttpd] decode_plus_to_space =
+  true`` setting. *In CouchDB 4.x, this is going to be an opt-out policy.*
+
+* :ghissue:`3724`: CouchDB now has new CSP settings for attachments and show/list
+  functions. This deprecates the old ``[csp] enable`` and ``[csp] header_value``
+  settings, replacing them with the new ``[csp] utils_enable`` and ``[csp]
+  utils_header_value`` settings respectively. In addition, new settings for
+  ``attachments_enable``, ``attachments_header_value``, ``showlist_enable`` and
+  ``showlist_header_value`` now are avialable. Documentation is in the ``default.ini``
```
Comment (rnewson) [inline src/whatsnew/3.2.rst:132]: typo at avialable

File: src/whatsnew/3.2.rst [hunk 6]
```diff
@@ -57,4 +113,140 @@ Features and Enhancements
...
+  is overloaded, due to a hard-coded 100ms timeout. CouchDB now calculates a more
+  reasonable timeout, based on the number of shards and the overall maximum fabric
+  request timeout limit, using a geometric series.
+
+Performance
+-----------
+
+* :ghissue:`3337`: Developer nodes now start faster when using the ``dev/run`` script.
+
+* :ghissue:`3366`: The monolithinc ``couch_server`` process has been sharded for
```
Comment (rnewson) [inline src/whatsnew/3.2.rst:146]: typo at monolithinc

File: src/whatsnew/3.2.rst [hunk 7]
```diff
@@ -57,4 +113,140 @@ Features and Enhancements
...
+  performance considerations. Previously, as a single ``gen_server``, the process would
```
Comment (rnewson) [inline src/whatsnew/3.2.rst:147]: s/considerations//

**Review by rnewson — CHANGES_REQUESTED** [review]

File: src/whatsnew/3.2.rst [hunk 8]
```diff
@@ -36,25 +92,161 @@ Features and Enhancements
...
+  ( :ghissue:`3370` )
+
+* :ghissue:`3208`: CouchDB now uses the latest ibrowse 4.4.2 client for the replicator.
+
+* :ghissue:`3600`, :ghissue:`3047`, :ghissue:`3019`: The default ``slack`` channel for
+  smoosh auto-compaction has been increased to a more reasonable value, reducing load
+  on systems that would have normally been idle in CouchDB 2.x (where no auto-compaction
+  daemon exists).
+
+* :ghissue:`3711`: Changes feeds may no longer rewind after shard moves, assumimg the
```
Comment (kocolosk) [inline src/whatsnew/3.2.rst:160]: typo: "assuming"

**Review by rnewson — APPROVED** [review]

**Comment by Terreii** [comment]
Sorry for waiting until now.

It seems like [3483](https://github.com/apache/couchdb/pull/3483) is missing.
Is it ok if I open a pull request to add it? Or do one of you write the release notes?

**Comment by nickva** [comment]
@Terreii definitely, make a docs PR and we'll add it. That's a very nice contribution, sorry for omitting it.

**Comment by Terreii** [comment]
Thanks, and no problem! It is after all a "strong release".

**Pull request merged**